dbcon/joblist/groupconcat.cpp

```diff
@@ -471,16 +471,15 @@ void GroupConcator::outputRow(std::ostringstream& oss, const rowgroup::Row& row)
                 else
                 {
                     int64_t intVal = row.getIntField(*i);
+
                     if (scale == 0)
                     {
                         oss << intVal;
                     }
                     else
                     {
-                        char buf[utils::MAXLENGTH8BYTES];
-                        dataconvert::DataConvert::decimalToString(intVal,
-                             scale, buf, sizeof(buf), types[*i]);
-                        oss << fixed << buf;
+                        long double dblVal = intVal / pow(10.0, (double)scale);
```

Comment:
This doesn't look right. I believe we should teach decimalToString to do the thing if it can't.

Reply:
I believe you can restore the the old code replacing pow() with a ref to a dict of 10^n values.

Reply:
Fixing pre-existing bugs in narrow decimals should be outside the scope of MCOL-641.

For narrow decimals, performing a floating point division for further processing appears to be a common theme. As an example, see RowAggregation::doSum: here for narrow decimals, we are again using `valIn /= pow(10.0, scale);`. I am sure there will be a lot of other places as well that will need changes. This commit simply reverts back changes performed to narrow decimals to the original state.

Reply:
I would disagree given that we are going to rewrite decimal processing to some degree. It is a very good point that the change must be introduced throughout the whole code base so I'll create an issue to replace all pow(10.0, (double)scale) expressions with a lookup in a values dictionary where the scale < 19 and assign it to you.